src/utils/chat.ts

```diff
@@ -0,0 +1,43 @@
+import { PostData } from '@subsocial/api/types'
+
+export function isMessageBlocked(
+  message: PostData | undefined | null,
+  blockedData: {
+    addresses: string[]
+    contentIds: string[]
+    postIds: string[]
+  }
+) {
+  if (!message) return false
+
+  const { addresses, contentIds, postIds } = blockedData
+  const { id, struct } = message
+  const contentId = struct.contentId
+  const owner = struct.ownerId
+
+  return (
+    addresses.includes(owner) ||
+    contentIds.includes(contentId ?? '') ||
+    postIds.includes(id)
+  )
+}
+
+export function getLastMessageId(
+  commentIds: string[] | undefined,
+  blockedIds: string[] | undefined
+) {
+  if (!commentIds) return undefined
+  let idx = commentIds.length - 1
+  while (true) {
+    if (idx < 0) {
+      return undefined
+    }
+
+    const id = commentIds[idx]
+    if (!blockedIds?.includes(id)) {
```

Comment:
I think better if you create Sets for blocked data after loading and use Sets in all places